pystac_client/item_search.py

```diff
@@ -683,6 +683,7 @@ def get_all_items_as_dict(self) -> Dict[str, Any]:
             "get_all_items_as_dict is deprecated, use get_items or"
             " get_item_collections instead",
             DeprecationWarning,
+            stacklevel=2,
```

Comment:
The warning was pointing back to this function, rather than user code. I *think* this is because of the `lru_cache`, but I might be wrong. We should verify this and add it to the other deprecated functions if necessary.